### PR TITLE
Support nested mixes

### LINKED ISSUE #16: Support nested mixes
if `instance.buildClassName` exec else build by b

### PR DESCRIPTION
**Pull request by awinogradov**
Close #16

### CODE REVIEW

File: src/Bem.js [hunk 1]
```diff
@@ -91,18 +92,42 @@ export default function({ preset, naming }) {
...
+                                    mixedEntities[k].mods = {
+                                        ...getMods({ ...mixed, ...mixedEntities[k] }),
+                                        ...mixed.mods
+                                    };
+                                else mixedEntities[k] = mixed;
+                            },
+                            resolveMixes = entities => {
+                                for(let entity of [].concat(entities))
+                                    if(entity) {
+                                        let mixed = entity;
```
Comment (veged) [inline src/Bem.js]: we can use `entity` and avoid adding new variable `mixed`

File: src/Bem.js [hunk 2]
```diff
@@ -91,18 +92,42 @@ export default function({ preset, naming }) {
...
+
+                                        if(typeof mixed.type === 'function') {
+                                            const props = mixed.props,
+                                                inst = new entity.type(props);
+
+                                            mixed = {
+                                                block : inst.block,
+                                                elem : inst.elem,
+                                                mods : inst.mods(props),
+                                                mix : inst.mix(props)
```
Comment (Yeti-or) [inline src/Bem.js]: addMix

File: src/Bem.js [hunk 3]
```diff
@@ -91,18 +92,42 @@ export default function({ preset, naming }) {
...
-
-                            const k = `${mixed.block}$${mixed.elem}`;
-
-                            if(uniq[k]) uniq[k].mods = { ...getMods({ ...mixed, ...uniq[k] }), ...mixed.mods };
-                            else uniq[k] = mixed;
-
-                            return uniq;
-                        }, {});
+                        const mixedEntities = {},
+                            resolveEntity = mixed => {
```
Comment (veged) [inline src/Bem.js]: resolveEntity -> resolveMixed

File: src/Bem.js [hunk 4]
```diff
@@ -91,18 +92,42 @@ export default function({ preset, naming }) {
...
+                                const k = `${mixed.block}$${mixed.elem}`;
+
+                                if(mixedEntities[k])
+                                    mixedEntities[k].mods = {
+                                        ...getMods({ ...mixed, ...mixedEntities[k] }),
+                                        ...mixed.mods
+                                    };
+                                else mixedEntities[k] = mixed;
+                            },
+                            resolveMixes = entities => {
```
Comment (veged) [inline src/Bem.js]: entities -> mixes

Comment (awinogradov) [reply src/Bem.js]: Ok ;)

**Comment by awinogradov** [comment]
JS mixes https://github.com/bem/bem-react-core/issues/178

**Pull request merged**